yarn-project/simulator/src/avm/avm_execution_environment.ts

```diff
@@ -35,7 +35,7 @@ export class AvmExecutionEnvironment {
       functionSelector,
       this.contractCallDepth.add(Fr.ONE),
       this.transactionFee,
-      this.header,
+      this.historicalHeader,
```

Comment:
I think this is supposed to be the current header and not the historical one. IIRC public will not have access to historical headers, since that would force nodes to be archivers.

Reply:
OTOH, I'm pretty sure header is currently unused and could be removed (we only keep it for the potential future HEADERMEMBER). When I added it, it was for interoperability with ACVM: https://github.com/AztecProtocol/aztec-packages/pull/5592

@dbanks12

Reply:
https://github.com/AztecProtocol/aztec-packages/blob/2570b59aee921a23841e135bde9b85fd67b442e6/yarn-project/simulator/src/public/public_processor.ts#L57

I updated it based on the public processor factory - it instantiates the avm executor with a variable called historical header - should this be updated instead?

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8990